File: 351004/Brazhalovich/Lab_1/src/main/java/org/example/newsapi/controller/MarkerController.java
package org.example.newsapi.controller;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.example.newsapi.dto.request.MarkerRequestTo;
import org.example.newsapi.dto.response.MarkerResponseTo;
import org.example.newsapi.service.MarkerService;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1.0/markers")
@RequiredArgsConstructor
public class MarkerController {
    private final MarkerService markerService;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public MarkerResponseTo create(@RequestBody @Valid MarkerRequestTo request) {
        return markerService.create(request);
    }

    @GetMapping
    public List<MarkerResponseTo> getAll() {
        return markerService.findAll();
    }

    @GetMapping("/{id}")
    public MarkerResponseTo getById(@PathVariable Long id) {
        return markerService.findById(id);
    }

    @PutMapping("/{id}")
    public MarkerResponseTo update(@PathVariable Long id, @RequestBody @Valid MarkerRequestTo request) {
        return markerService.update(id, request);
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void delete(@PathVariable Long id) {
        markerService.delete(id);
    }
}

File: 351003/brezgunov/distcomp/src/main/kotlin/com/distcomp/controller/MarkerController.kt
package com.distcomp.controller

import com.distcomp.dto.marker.MarkerRequestTo
import com.distcomp.dto.marker.MarkerResponseTo
import com.distcomp.service.MarkerService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/{version}/markers")
class MarkerController(
    private val markerService: MarkerService
) {
    @PostMapping(version = "1.0")
    @ResponseStatus(HttpStatus.CREATED)
    fun createMarker(@RequestBody markerRequestTo: MarkerRequestTo): MarkerResponseTo {
        return markerService.createMarker(markerRequestTo)
    }

    @GetMapping("{id}")
    fun readMarkerById(@PathVariable("id") id: Long): MarkerResponseTo {
        return markerService.readMarkerById(id)
    }

    @GetMapping
    fun findAll(): List<MarkerResponseTo> {
        return markerService.readAll()
    }

    @PutMapping(path = ["/{id}", ""], version = "1.0")
    @ResponseStatus(HttpStatus.OK)
    fun updateMarker(
        @RequestBody markerRequestTo: MarkerRequestTo,
        @PathVariable("id") id: Long?
    ): MarkerResponseTo {
        return markerService.updateMarker(markerRequestTo, id)
    }

    @DeleteMapping(path = ["/{id}"], version = "1.0")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteMarker(@PathVariable("id") id: Long) {
        markerService.removeMarkerById(id)
    }
}